ui/src/hooks/useKeycloak.tsx

```diff
@@ -37,86 +37,119 @@ const useKeycloak = (adapter: KeycloakAuthAdapter, loginRequired?: boolean) : Au
+        setUserId(undefined);
+        setTokenExpired(undefined);
+        const message = e?.error_description ? e.error_description : 'Failed to authenticate';
+        setError(message);
+        setIsError(true);
+        setInitializing(false);
+        setAuthenticated(false);
+        setAuthenticating(false);
+        setLoggingOut(false);
+      };
+
+      keycloak.onTokenExpired = () => {
+        console.log('Keycloak onTokenExpired');
+        keycloak.updateToken(30).catch(() => {
+          console.error('Failed to update token');
           keycloak.clearToken();
           setUserId(undefined);
-          setTokenExpired(undefined);
-          const message = (e?.error_description)?e.error_description:'Failed to authenticate';
-          setError(message);
-          setIsError(true);
-          setInitializing(false);
-          setAuthenticated (false);
+          setTokenExpired(true);
+          setAuthenticated(false);
           setAuthenticating(false);
-          setLogingOut(false);
-        };
-        keycloak.onTokenExpired = () => {
-          keycloak
-            .updateToken(30)
-            .catch(() => {
-              keycloak.clearToken();
-              setUserId(undefined);
-              setTokenExpired(true);
-              setAuthenticated(false);
-              setAuthenticating(false);
-              setLogingOut(false);
-            });
-        };
-        const initOptions = adapter.initOptions?{
-          ...adapter.initOptions,
-          onLoad: loginRequired?'login-required':adapter.initOptions.onLoad,
-          checkLoginIframe: !!adapter.initOptions.checkLoginIframe && !window.location.host.startsWith('localhost') // avoid CORS error with UI running on localhost with Firefox
-        }:{};
-        keycloak
-          .init(initOptions)
-          .catch(() => {
-            setError('Failed to initialize authentication');
-            setIsError(true);
-            setInitialized(false);
-            setInitializing (false);
-            setAuthenticating(false);
-            setLogingOut(false);
-          });
-      } catch (e) { // if keycloak script url return unexpected content
-        setError('Failed to initialize authentication');
-        setInitialized(false);
-        setInitializing(false);
-        setAuthenticating(false);
-      }
-    } else {
-      //throw new Error("keycloak cannot be initialized due to missing/uncomplete config!");
+          setLoggingOut(false);
+        });
+      };
+
+      const initOptions = adapter.initOptions
+        ? {
+            ...adapter.initOptions,
+          }
+        : {};
+
+      await keycloak.init(initOptions);
+    } catch (e) {
+      console.error('Keycloak initialization error', e);
       setError('Failed to initialize authentication');
       setInitialized(false);
       setInitializing(false);
       setAuthenticating(false);
     }
   };
 
+
+  const runPreflightCheck = async (): Promise<boolean> => {
```

Comment:
Please note, that there is already another place where the service calls are typically handled -> https://github.com/marmotgraph/marmotgraph-search/blob/main/ui/src/services/api.ts we might want to be consistent and move the call to the new endpoint to the others.